docs/application/dotnet/guides/nui/Palette.md

```diff
@@ -0,0 +1,102 @@
+        {
+            //Exception Handling.
+        }
+    }
+    
+    // Generate a Palette asynchronously
+    public async Task PaletteGenerateAsync()
+    {
+        PixelBuffer imgBitmap = ImageLoading.LoadImageFromFile(_imgPath);
+        try
+        {
+            Palette palette = await Palette.GenerateAsync(imgBitmap);
+        }
+        catch (ArgumentNullException e)
+        {
+            //Exception Handling.
+        }
+    }
+    ```
+ 
+ 3. Set a region of the pixelBuffer to be used exclusively when calculating the palette. (Optional)
+ 
+    ```csharp
+    Rectangle rect = new Rectangle(0, 0, 100, 100);
+    ...
+    Palette palette = Palette.Generate(imgBitmap, rect);
+    ```
+
+## Extract color profiles
+
+Colors extracted from the bitmap image are scored against each profile based on saturation, luminance, and population (number of pixels in the bitmap represented by the color).
+For each profile, the color with the best score defines that color profile for the given image.
+Palette object contains 16 primary colors from a given image.
+
+The palette library attempts to extract the following six color profiles:
+
+- Light Vibrant
+- Vibrant
+- Dark Vibrant
+- Light Muted
+- Muted
+- Dark Muted
+
+Each of Palette’s get<Profile>Color() methods returns the color in the palette associated with that particular profile,
+where <Profile> is replaced by the name of one of the six color profiles.
+For example, the method to get the Dark Vibrant color profile is getDarkVibrantColor().
+Since not all images will contain all color profiles, you must also provide a default color to return.
+
+![Palette](./media/Palette.png)
+    
+To access all colors in a palette, the getSwatches() method returns a list of all swatches generated from an image, including the standard six color profiles.
+    
+
+## Use swatches to create color schemes
+
+The Palette class also generates Palette.Swatch objects for each color profile.
+Palette.Swatch objects contain the associated color for that profile, as well as the color’s population in pixels.
+
+Swatches have additional methods for accessing more information about the color profile, such as HSL values and pixel population.
+You can use swatches to help create more comprehensive color schemes and app themes using the getBodyTextColor() and getTitleTextColor() methods.
+These methods return colors appropriate for use over the swatch’s color.
+
+Each of Palette’s get<Profile>Swatch() methods returns the swatch associated with that particular profile, where <Profile> is replaced by the name of one of the six color profiles.
+Although the palette’s get<Profile>Swatch() methods do not require default value parameters, they return null if that particular profile does not exist in the image.
+Therefore, you should check that a swatch is not null before using it. For example, the following code gets the title text color from a palette if the Vibrant swatch is not null
```

Comment:
```suggestion
- Each of Palettes `get<Profile>Swatch()` returns the swatch associated with that particular profile, where <Profile> is replaced by the name of one of the six color profiles.
- The palettes `get<Profile>Swatch()` does not require default value parameters, the method returns null if a particular profile does not exist in the image. Therefore, you should check that a swatch is not null before using it. For example, the following code gets the title text color from a palette, if the vibrant swatch is not null. 
```


Code seems to be missing, the last line talks about the following code. Please verify.